dbcon/dmlpackage/insertdmlpackage.cpp

```diff
@@ -100,15 +100,50 @@ int InsertDMLPackage::read(messageqcpp::ByteStream& bytestream)
     bytestream >> logending;
     fLogending = (logending != 0);
 
+    bytestream >> fTableOid;
+    bytestream >> reinterpret_cast<messageqcpp::ByteStream::byte&>(fIsInsertSelect);
+    bytestream >> reinterpret_cast<messageqcpp::ByteStream::byte&>(fIsBatchInsert);
+    bytestream >> reinterpret_cast<messageqcpp::ByteStream::byte&>(fIsAutocommitOn);
+
     fTable = new DMLTable();
     retval = fTable->read(bytestream);
-    bytestream >> fTableOid;
-    bytestream >> reinterpret_cast< messageqcpp::ByteStream::byte&>(fIsInsertSelect);
-    bytestream >> reinterpret_cast< messageqcpp::ByteStream::byte&>(fIsBatchInsert);
-    bytestream >> reinterpret_cast< messageqcpp::ByteStream::byte&>(fIsAutocommitOn);
     return retval;
 }
 
+void InsertDMLPackage::readMetaData(messageqcpp::ByteStream& bytestream)
+{
+    messageqcpp::ByteStream::quadbyte session_id;
+    bytestream >> session_id;
+    fSessionID = session_id;
+    bytestream >> fUuid;
+
+    std::string dmlStatement;
+    bytestream >> fDMLStatement;
+    bytestream >> fSQLStatement;
+    bytestream >> fSchemaName;
+    bytestream >> fTimeZone;
+    uint8_t logging;
+    bytestream >> logging;
+    fLogging = (logging != 0);
+    uint8_t logending;
+    bytestream >> logending;
+    fLogending = (logending != 0);
+
+    bytestream >> fTableOid;
+    bytestream >> reinterpret_cast<messageqcpp::ByteStream::byte&>(fIsInsertSelect);
+    bytestream >> reinterpret_cast<messageqcpp::ByteStream::byte&>(fIsBatchInsert);
+    bytestream >> reinterpret_cast<messageqcpp::ByteStream::byte&>(fIsAutocommitOn);
+
+    fTable = new DMLTable();
+    fTable->readMetaData(bytestream);
+}
+
```

Comment:
Is readMetadata() different than read()?  I don't see a difference except for the return val, am I just missing it?

Reply:
It doesn't deser an array of records values from the stream.

Reply:
Exactly. This is the main reason to add this method; to avoid deserializing the row data containing the vectors of column values, which is wasteful in dmlproc as we only need a few metadata fields from the dml package there, and not the actual row data.